### PR TITLE
gpu: NVIDIA GPU initrd/image build

### PR DESCRIPTION
**Pull request by zvonkok**
With each release make sure we ship a GPU enabled rootfs/initrd

Fixes: #6554
Depends: ~~#9480~~, ~~#9584~~

### CODE REVIEW

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:467]: This was removed a few commits ago, and I think you mistakenly re-added it as part of a rebase.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:1147]: Extra line added.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:518]: We're not prepared for MEASURED_ROOTFS using initrd.

The work I've done was focusing on the image side, so I´d rather suggest to not export this here, and then re-add it whenever working on the initrd support for MEASURED_ROOTFS.

File: tools/packaging/kata-deploy/local-build/Makefile
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/Makefile:166]: Why adding the dependencies commented out?
Ditto for all the other targets added.

**Review by fidencio — CHANGES_REQUESTED** [review]
@zvonkok, I sincerely lack the proper knowledge to verify that what you're doing inside the nvidia rootfs-builder is correct or not.  So, I will leave that for your discretion.

I've gone through the common files and spotted a few things that must be changed before having this one merged.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (zvonkok) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:1147]: Yes, otherwise we would have 
```
kernel-confidential) install_kernel_confidential ;;

kernel-dragonball-experimental) install_kernel_dragonball_experimental ;;
kernel-nvidia-gpu-dragonball-experimental) install_kernel_nvidia_gpu_dragonball_experimental ;;

kernel-nvidia-gpu) install_kernel_nvidia_gpu ;;
```

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (zvonkok) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:518]: Ok commented out and left as TODO. I need this for initrds as well.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (zvonkok) [reply tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:467]: Removed.

File: tools/packaging/kata-deploy/local-build/Makefile
Comment (zvonkok) [reply tools/packaging/kata-deploy/local-build/Makefile:166]: Local builds :) Forgot to add them in

**Comment by zvonkok** [comment]
@fidencio all green PTAL

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:512]: Here MEASURED_ROOTFS=yes will work.

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (fidencio) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:522]: Here, it won't, as we don't have the machinery.

**Review by fidencio — COMMENTED** [review]
@zvonkok, one last question for you.

MEASURED_ROOTFS=yes would work no problem for the rootfs image.  Do you prefer keeping it off and enable everything at the same once you have the rootfs initrd situation figured out?

**Comment by fidencio** [comment]
> MEASURED_ROOTFS=yes would work no problem for the rootfs image. Do you prefer keeping it off and enable everything at the same once you have the rootfs initrd situation figured out?

@zvonkok told me privately that he'll get the MEASURED_ROOTFS=yes for both cases at the same time, in future PRs.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @zvonkok!

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh:229]: There is a mix of spaces and tabs here.

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:134]: Nit: Mix of tabs and spaces here.

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:165]: nit: Indentation issue

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:178]: I'm not sure if three lines of white space is needed/helpful?

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:183]: Unnecessarily new line here?

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:199]: nit: Mix of spaces and tabs here

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:297]: Mix of spaces and tabs here too

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:317]: nit: Inconsistant indentation

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:357]: Inconsistent indentation here

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (stevenhorsman) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:365]: Unnecessary new lines?

File: tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh
Comment (stevenhorsman) [inline tools/packaging/kata-deploy/local-build/kata-deploy-binaries.sh:507]: nit: Type on `Install`

**Review by stevenhorsman — COMMENTED** [review]
Unfortunately I don't have my knowledge on the gpu specifics to know whether it's correct, so I've ended up doing more of a code style review, which did throw up a few issues, but I appreciate it's of super high value.

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh
Comment (zvonkok) [reply tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh:25]: Aye, will update. @stevenhorsman The bash linting is also overdue will create a separate PR

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh
Comment (zvonkok) [reply tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh:29]: No will remove.

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh
Comment (zvonkok) [reply tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh:27]: Good eyes :)

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh
Comment (Apokleos) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_rootfs.sh:32]: ah..., do you have a plan to replace this repo with the official one ?

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (Apokleos) [inline tools/osbuilder/rootfs-builder/rootfs.sh:832]: According to this line 832 here, it seems there have been some changes when creating the GPU rootfs, right? It needs ENV VARIANT to indicate it is a gpu branch and do setup nvidia gpu rootfs with two stages.  what I miss here? I can't find how to set this ENV.

**Review by Apokleos — COMMENTED** [review]
Hi Zvonko, I am testing this PR, but I notice that this PR will be blocked by another issue about `Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*']`
IMO, we need first address this problem.

**Comment by Apokleos** [comment]
```shell
   Compiling rustls-webpki v0.102.5
error: failed to run custom build command for `loopdev v0.5.0 (https://github.com/mdaffin/loopdev?rev=c9f91e8f0326ce8a3364ac911e81eb32328a5f27#c9f91e8f)`

Caused by:
  process didn't exit successfully: `/kata-containers/src/agent/target/release/build/loopdev-61866741446c5cc7/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at /opt/cargo/registry/src/index.crates.io-6f17d22bba15001f/bindgen-0.63.0/./lib.rs:2338:31:
  Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
make: *** [Makefile:137: target/x86_64-unknown-linux-musl/release/kata-agent] Error 101
```
Related issues:
https://github.com/kata-containers/kata-containers/pull/9407
https://github.com/kata-containers/kata-containers/pull/10377
one more less related, but they are with the same problem:
https://github.com/kata-containers/kata-containers/pull/10436

File: tools/osbuilder/rootfs-builder/rootfs.sh
Comment (zvonkok) [reply tools/osbuilder/rootfs-builder/rootfs.sh:832]: This is set when executing rootfs.sh with the proper VARIANT. See the modified call to the rootfs.sh script.

**Comment by zvonkok** [comment]
@Apokleos Now that we merged: https://github.com/kata-containers/kata-containers/pull/10377 we can rebase this and the build error should be gone.

**Comment by zvonkok** [comment]
@Apokleos All green, PTAL

File: tools/packaging/kata-deploy/local-build/Makefile
Comment (sprt) [inline tools/packaging/kata-deploy/local-build/Makefile:83]: nit: I'd put this at the top of the file.

File: tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh
Comment (sprt) [inline tools/osbuilder/rootfs-builder/nvidia/nvidia_chroot.sh:42]: nit: I wouldn't pipe to /dev/null so we're consistent with other scripts in the repo.

**Review by sprt — APPROVED** [review]
LGTM, just a couple minor nits.

**Pull request merged**
